must-gather/collection-scripts/cephfilesystems.sh

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+resource="cephfilesystems"
+
+base_collection_path=$1
+namespace=$2
+if [ "${base_collection_path}" = "" ];then
+    echo "Base collection path for ${resource} is not passed. Exiting."
+    exit 0
+fi
+
+if [ "${namespace}" = "" ];then
+    echo "Namespace for ${resource} is not passed. Exiting."
+    exit 0
+fi
+
+# removing previous collector files
+rm -rf ${resource}-collector.sh
+
+api_group="ceph.rook.io"
+if [ "${api_group}" != "" ]; then 
+    base_collection_path="${base_collection_path}/${api_group}"
+fi
+
+base_collection_path="${base_collection_path}/${resource}"
+mkdir -p "${base_collection_path}"
+
+
+# Collection jsonpath
+collection_jsonpath="{range .items[*]}"
+collection_jsonpath="${collection_jsonpath}echo ' -> Fetching dump for {@.metadata.name} cephfilesystem';"
+collection_jsonpath="${collection_jsonpath}timeout 120 oc get -n ${namespace} ${resource} {@.metadata.name} -o yaml > "
+collection_jsonpath="${collection_jsonpath}${base_collection_path}/{@.metadata.name}.yaml;"
+collection_jsonpath="${collection_jsonpath}{end}"
```

Comment:
There's a *lot* of identical or extremely similar code in these files. Might this be easier to maintain if they shared a function library to accomplish these very similar tasks? For this one you'd really only need to pass "cephfilesystem" as an arg to differentiate it from other jsonpaths you're building.

Reply:
yup I was having it in my mind to convert those into functions.